From cf9e76b19482cf494798c24ffe904485a36a0b19 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@gentoo.org>
Date: Tue, 21 Feb 2012 19:42:22 +0530
Subject: [PATCH] [tests] Run gdb under libtool so that local libraries are used

If this isn't done, the system-wide glib is used for this test,
and that can cause test failures.

https://bugzilla.gnome.org/621368
---
 tests/run-assert-msg-test.sh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/tests/run-assert-msg-test.sh b/tests/run-assert-msg-test.sh
index 194481e..b6045d6 100755
--- a/tests/run-assert-msg-test.sh
+++ b/tests/run-assert-msg-test.sh
@@ -34,7 +34,7 @@ if [ -e ".libs/lt-$msg_test" ]; then
        msg_test="lt-$msg_test"
 fi
 echo_v "Running gdb on assert-msg-test"
-OUT=$(gdb --batch --ex run --ex "set print elements 0" --ex "print (char*) __glib_assert_msg" .libs/$msg_test 2> $error_out) || \
+OUT=$(libtool --mode=execute gdb --batch --ex run --ex "set print elements 0" --ex "print (char*) __glib_assert_msg" .libs/$msg_test 2> $error_out) || \
   fail "failed to run gdb"
 
 echo_v "Checking if assert message is in __glib_assert_msg"
-- 
1.7.3.4

